src/components/Inputs/input.module.scss

```diff
@@ -1438,6 +1447,93 @@
     &.input-stretch {
         width: 100%;
     }
+
+    &.isExpandable {
+        .expandable-wrapper {
+            position: relative;
+        }
+
+        .input-group {
+            position: absolute;
+            left: 0;
+            z-index: 10;
+
+            input {
+                width: 0;
+            }
+
+            .clear-icon-button {
+                right: -12%;
+            }
+        }
+
+        .expandable-thumb {
+            position: absolute;
+            left: -3px;
+        }
+
+        .overlay {
+            position: absolute;
+            left: 0;
+            top: 0;
+            height: 100%;
+            width: 200%;
+            z-index: -1;
+            background: $expandable-header-gradient;
+        }
+
+        input,
+        .input-group,
+        .overlay {
+            transition: all $motion-duration-fast $motion-easing-easeout;
+        }
+
+        // Set Rules
+        .input-group:hover > input,
+        input:focus,
+        input:not(:placeholder-shown) {
+            @each $size-name, $size-value in $input-size-map {
+                &.input-#{$size-name} {
+                    width: auto;
+                }
+            }
+
+            ~ .actionWrapper {
+                display: block;
+            }
+
+            ~ .expandable-thumb {
+                display: none;
+            }
+        }
+
+        // Unset Rules
+        @each $size-name, $size-value in $input-size-map {
+            &.input-#{$size-name} > .expandable-wrapper {
+                min-width: $size-value;
+                min-height: $size-value;
+            }
+
+            .input-group:not(:hover)
+                input.input-#{$size-name}:placeholder-shown {
+                min-width: $size-value;
+            }
+
+            .input-group:not(:hover)
+                input.input-#{$size-name}:placeholder-shown {
+                padding: 0;
+                overflow: hidden;
+
+                &::placeholder {
```

Comment:
This is unlikely to work in ie11... does it get polyfilled in compilation?
https://caniuse.com/?search=placeholder

Reply:
Additionally, the instant change to font-size seems to cause an unsightly overlap of text and search icon during the animation. You might want to rethink this approach.

Reply:
will add a polyfill for this.